lib/src/Base/Solver/LeastSquaresEquationsSolver.cxx

```diff
@@ -0,0 +1,142 @@
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "openturns/LeastSquaresEquationsSolver.hxx"
+#include "openturns/LeastSquaresProblem.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/OptimizationAlgorithm.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/SymbolicFunction.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class LeastSquaresEquationsSolver
+ *
+ * This class is an interface for the nonlinear LeastSquaresEquationsSolver
+ */
+
+CLASSNAMEINIT(LeastSquaresEquationsSolver)
+
+static const Factory<LeastSquaresEquationsSolver> Factory_LeastSquaresEquationsSolver;
+
+/* Parameter constructor */
+LeastSquaresEquationsSolver::LeastSquaresEquationsSolver(const Scalar absoluteError,
+                                                         const Scalar relativeError,
+                                                         const Scalar residualError,
+                                                         const UnsignedInteger maximumCallsNumber)
+  : SolverImplementation(absoluteError, relativeError, residualError, maximumCallsNumber)
+{
+  // LeastSquares problem
+  LeastSquaresProblem problem(SymbolicFunction("x", "x"));
+  solver_ = OptimizationAlgorithm::Build(problem);
+  solver_.setMaximumCallsNumber(maximumCallsNumber);
+  solver_.setMaximumAbsoluteError(absoluteError);
+  solver_.setMaximumRelativeError(relativeError);
+  solver_.setMaximumResidualError(residualError);
+}
+
+/* Virtual constructor */
+LeastSquaresEquationsSolver * LeastSquaresEquationsSolver::clone() const
+{
+  return new LeastSquaresEquationsSolver(*this);
+}
+
+/* String converter */
+String LeastSquaresEquationsSolver::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << LeastSquaresEquationsSolver::GetClassName()
+      << " derived from " << SolverImplementation::__repr__();
+  return oss;
+}
+
+void LeastSquaresEquationsSolver::setOptimizationAlgorithm(const OptimizationAlgorithm & algorithm)
+{
+  solver_ = algorithm;
+}
+
+OptimizationAlgorithm LeastSquaresEquationsSolver::getOptimizationAlgorithm() const
+{
+  return solver_;
+}
+
+/* Solve attempt to find one root to the system of non-linear equations function(x) = 0
+   given a starting point x with a least square optimization method.
+*/
+Point LeastSquaresEquationsSolver::solve(const Function & function,
+                                         const Point & startingPoint) const
+{ 
+  const Interval bounds;
```

Comment:
Shouldn't the bounds include the starting point? And what about the dimension of the default Interval wrt the dimension of the starting point or the input dimension of the function?
IMO, something like
```
const Interval bounds(Point(startingPoint.getDimension(), -SpecFunc::MaxScalar), Point(startingPoint.getDimension(), SpecFunc::MaxScalar));
```
would give more sense to these bounds.

Reply:
`Interval(0)` is used by the `OptimizationProblem` thus it is in coherency with the rest of the code
Also we check that `Interval.getBounds.getSize` is cimpatible with `objective.getInputFunction`

Reply:
these are bounds of dimension 0 which means no bounds for the optim solver

Reply:
Correct, so it is ok for me!